java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -88,13 +97,68 @@ public JsonFileReader(File inputFile, BufferAllocator allocator) throws JsonPars
     this.parser = jsonFactory.createParser(inputFile);
   }
 
+  @Override
+  public Dictionary lookup(long id) {
+    if (initialized) {
+      return dictionaries.get(id);
+    } else {
+      return null;
```

Comment:
Yeah, I agree on throwing an exception.  This was taken from ArrowReader, I'll make the change there also to be consistent unless you object.